synapse/api/auth.py

```diff
@@ -186,7 +186,7 @@ async def get_user_by_req(
             ip_addr = self.hs.get_ip_from_request(request)
             user_agent = request.requestHeaders.getRawHeaders(
                 b"User-Agent", default=[b""]
-            )[0].decode("ascii", "surrogateescape")
+            )[0].decode("ascii", "replace")
```

Comment:
I don't know if using `"ascii"` her is correct?

Reply:
Should we have a `hs.get_user_agent_from_request` method like `get_ip_from_request` to consolidate all this logic into one spot at least?

Reply:
Good point well made